cmd/fastwalk.go

```diff
@@ -44,16 +43,7 @@ var errSkipFile = errors.New("fastwalk: skip this file")
 //   * fastWalk can follow symlinks if walkFn returns the TraverseLink
 //     sentinel error. It is the walkFn's responsibility to prevent
 //     fastWalk from going into symlink cycles.
-func fastWalk(root string, walkFn func(path string, typ os.FileMode) error) error {
-	// TODO(bradfitz): make numWorkers configurable? We used a
-	// minimum of 4 to give the kernel more info about multiple
-	// things we want, in hopes its I/O scheduling can take
-	// advantage of that. Hopefully most are in cache. Maybe 4 is
-	// even too low of a minimum. Profile more.
-	numWorkers := 4
-	if n := runtime.NumCPU(); n > numWorkers {
-		numWorkers = n
-	}
+func fastWalk(root string, nworkers int, doneCh <-chan struct{}, walkFn func(path string, typ os.FileMode) error) error {
```

Comment:
```suggestion
func fastWalk(root string, doneCh <-chan struct{}, items chan<- Item) error {
    nworkers := cap(items)
```

And have a `Result chan<- error` inside each Item. Each worker can have its own, so no allocations needed. The channel should have a length one so the result can always be returned without blocking.

When an item is sent, it will be picked up and the result is sent on the channel of the Item. This way `items` can be buffered and potentially (but not necessarily) there can even be several picking up items.

This removes the need for unbuffered channels and still allows individual feedback.